src/Generator/PropertyGenerator.php

```diff
@@ -220,14 +220,19 @@ public function generate()
             }
             $output .= $this->indentation . 'const ' . $name . ' = '
                 . ($defaultValue !== null ? $defaultValue->generate() : 'null;');
-        } else {
-            $output .= $this->indentation
-                . $this->getVisibility()
-                . ($this->isStatic() ? ' static' : '')
-                . ' $' . $name . ' = '
-                . ($defaultValue !== null ? $defaultValue->generate() : 'null;');
+
+            return $output;
         }
 
+        $output .= $this->indentation . $this->getVisibility() . ($this->isStatic() ? ' static' : '') . ' $' . $name;
+
+        if ($this->defaultValue instanceof PropertyValueGenerator &&
+            $this->defaultValue->getType() === ValueGenerator::TYPE_OMIT) {
```

Comment:
`$this->defaultValue->omitType()` instead - that gets rid of the ugly constant too ;-)